scene/animation/animation_player.cpp

```diff
@@ -1072,25 +1072,31 @@ void AnimationPlayer::_animation_process2(double p_delta, bool p_started) {
 	accum_pass++;
 
 	bool seeked = c.seeked; // The animation may be changed during process, so it is safer that the state is changed before process.
+	float blend = 1.0; // First animation we play at 100% blend
+
 	if (p_delta != 0) {
 		c.seeked = false;
 	}
 
-	_animation_process_data(c.current, p_delta, 1.0f, seeked, p_started);
-
-	List<Blend>::Element *prev = nullptr;
-	for (List<Blend>::Element *E = c.blend.back(); E; E = prev) {
+	List<Blend>::Element *next = NULL;
+	for (List<Blend>::Element *E = c.blend.front(); E; E = next) {
 		Blend &b = E->get();
-		float blend = b.blend_left / b.blend_time;
+		_animation_process_data(b.data, p_delta, blend, false, false);
+		blend = 1.0 - b.blend_left / b.blend_time; // This is how much to blend the NEXT animation
 		b.blend_left -= Math::absf(speed_scale * p_delta);
-		prev = E->prev();
+		next = E->next();
 		if (b.blend_left < 0) {
-			c.blend.erase(E);
+			// If the blend of this has finished, we need to remove ALL the previous blends
+			List<Blend>::Element *prev;
+			while (E) {
+				prev = E->prev();
+				c.blend.erase(E);
+				E = prev;
+			}
 		}
-		// The effect of animation changes during blending is unknown...
-		// In that case, we recommends to use method call mode "deferred", not "immediate".
-		_animation_process_data(b.data, p_delta, blend, false, false);
 	}
+
+	_animation_process_data(c.current, p_delta, blend, seeked && p_delta != 0, p_started);
```

Comment:
```suggestion
	_animation_process_data(c.current, p_delta, blend, seeked, p_started);
```
p_delta ! = 0 already has no meaning for seeking.

Reply:
Yeah, I wasn't 100% sure if that check should be there.  That's just how the code was before (this PR is very old).